desktop_version/src/VFormat.c

```diff
@@ -0,0 +1,463 @@
+                    return;
+                }
+
+                size_t flag_len = next_stop - cursor;
+
+                if (first_iter)
+                {
+                    name_len = flag_len;
+
+                    first_iter = false;
+                }
+                else if (flag_len == 5 && SDL_memcmp(cursor, "wordy", 5) == 0)
+                {
+                    flag_wordy = true;
+                }
+                else if (flag_len >= 8 && SDL_memcmp(cursor, "digits=", 7) == 0)
+                {
+                    /* strtol stops on the first non-digit anyway, so... */
+                    flag_digits = SDL_strtol(cursor + 7, NULL, 10);
+                }
+                else if (flag_len == 6 && SDL_memcmp(cursor, "spaces", 6) == 0)
+                {
+                    flag_spaces = true;
+                }
+                else if (flag_len == 5 && SDL_memcmp(cursor, "upper", 5) == 0)
+                {
+                    flag_upper = true;
+                }
+
+                cursor = next_stop + 1;
+            }
+            while (next_stop[0] != '}');
+
+            /* Now look up the name in the arguments.
+             * The arguments index string is comma-separated,
+             * each argument is name:type. */
+            va_list args_copy;
+            va_copy(args_copy, args);
+
+            const char* args_index_cursor = args_index;
+
+            bool match = false;
+            do
+            {
+                while (args_index_cursor[0] == ' ')
+                {
+                    /* Skip spaces between args */
+                    args_index_cursor++;
+                }
+
+                const char* next_comma = SDL_strchr(args_index_cursor, ',');
+                const char* next_colon = SDL_strchr(args_index_cursor, ':');
+                if (next_comma == NULL)
+                {
+                    next_comma = SDL_strchr(args_index_cursor, '\0');
+                }
+
+                if (next_colon == NULL || next_colon > next_comma)
+                {
+                    break;
+                }
+
+                const char* arg_name = args_index_cursor;
+                size_t arg_name_len = next_colon - arg_name;
+                const char* arg_type = next_colon + 1;
+                size_t arg_type_len = next_comma - arg_type;
+
+                strip_whitespace(&arg_name, &arg_name_len);
+                strip_whitespace(&arg_type, &arg_type_len);
+
+                match = (arg_name_len == name_len && SDL_memcmp(arg_name, name, name_len) == 0);
+
+                if (arg_type_len == 3 && SDL_memcmp(arg_type, "int", 3) == 0)
+                {
+                    int value = va_arg(args_copy, int);
+
+                    if (match)
+                    {
+                        if (flag_wordy)
+                        {
+                            char* number = HELP_number_words(value);
```

Comment:
Just a note, for me more than anything (which you don't need to worry about): While it would be ideal for this to check for `NULL` and set a safe value on a failed alloc (and for the `HELP_number_words` wrapper to do the same thing), I was already planning on changing `help.number_words` to take in a buffer and be used with stack-allocated buffers, so that'd be a bit moot. (And this PR will help transition that along anyway :stuck_out_tongue:)

Reply:
In the localization branch, even the trinket counts in the save box and on the Game Complete screen should probably be VFormat'ted, so translators can choose whether to capitalize the first letter there. So VFormat would be the only place which calls `help.number_words`.